ext/opcache/ZendAccelerator.c

```diff
@@ -2079,6 +2081,24 @@ zend_op_array *persistent_compile_file(zend_file_handle *file_handle, int type)
 		}
 	}
 
+	/* If this script is part of a package, make sure the package declaration did not change. */
+	if (persistent_script && persistent_script->package &&
```

Comment:
Do I understand correctly in this block, if `package_declare()` changes, then the script gets re-compiled?

The case of composer is pretty straightforward, as the package declares would be generated into `vendor/autoload.php` or a subsequent file.

But technically it is possible to call `package_declare` dynamically with values that change every time the declaration is executed: `package_declare(['name' => 'foo/bar', 'declares' => ['strict_types' => random_int(0, 1)]]);`

Reply:
That's correct. Things like ticks or strict_types influence the way code is compiled, so we can't reuse the compilation if the package declaration changed.

It would be possible to instead cache multiple versions of the same file and chose between them. However, in practice the package declaration will never change (or if it changes, then the old version will never be used again), as such it is more efficient to only cache one version and verify that things are still the same.